junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/MethodArgumentsProvider.java

```diff
@@ -37,8 +38,10 @@ public void accept(MethodSource annotation) {
 		Object testInstance = context.getTestInstance().orElse(null);
 		// @formatter:off
 		return Arrays.stream(methodNames)
+				.filter(Objects::nonNull)
```

Comment:
Elements of an array supplied via an annotation can never be null according to the JLS. The code otherwise wouldn't compile.

So you can simply remove the `.filter(Objects::nonNull)` line.